src/components/settings/SettingsLMSTab/LMSConfigs/utils.tsx

```diff
@@ -0,0 +1,177 @@
+        }
+      }
+      oauthUrl = `${currentFormFields.blackboardBaseUrl}/learn/api/public/v1/oauth2/authorizationcode?`
+        + `redirect_uri=${BLACKBOARD_OAUTH_REDIRECT_URL}&scope=read%20write%20delete%20offline&`
+        + `response_type=code&client_id=${appKey}&state=${configUuid}`;
+    }
+    else {
+      oauthUrl =
+        `${currentFormFields.canvasBaseUrl}/login/oauth2/auth?client_id=${currentFormFields.clientId}&` +
+        `state=${currentFormFields.uuid}&response_type=code&` +
+        `redirect_uri=${CANVAS_OAUTH_REDIRECT_URL}`;
+    }
+    // Open the oauth window for the user	
+    window.open(oauthUrl);
+    dispatch?.(setWorkflowStateAction(WAITING_FOR_ASYNC_OPERATION, true));
+  }
+  return currentFormFields;
+}
+
+export async function afterSubmitHelper(
+  lmsType: string,
+  formFields: any,
+  channelMap: { [key: string]: {[key: string]: any }},
+  errHandler: FormWorkflowErrorHandler | undefined,
+  dispatch: any) {
+  if (formFields?.id) {
+    let err = "";
+    try {
+      const response = await channelMap[lmsType].fetch(formFields.id);
+      if (response.data.refresh_token) {
+        dispatch?.(
+          setWorkflowStateAction(WAITING_FOR_ASYNC_OPERATION, false)
+        );
+        return true;
+      }
+    } catch (error) {
+      err = handleErrors(error);
+    }
+    if (err) {
+      errHandler?.(err);
+      return false;
+    }
+  }
+  return false;
+}
+
+export async function onTimeoutHelper(
+  dispatch: any) {
+  dispatch?.(setWorkflowStateAction(WAITING_FOR_ASYNC_OPERATION, false));
+  dispatch?.(setWorkflowStateAction(LMS_AUTHORIZATION_FAILED, true));
+}
+
+export async function handleSaveHelper(
+  transformedConfig: ConfigSnakeCase,
+  existingData: any,
+  formFields: ConfigCamelCase,
+  onSubmit: Function,
+  lmsType: string,
+  channelMap: { [key: string]: {[key: string]: any }},
+  errHandler: (errMsg: string) => void) {
+  let err = "";
+  if (formFields.id) {
+    try {
+      transformedConfig.active = existingData.active;
+      const response = await channelMap[lmsType].update(transformedConfig, existingData.id);
+      onSubmit(formFields);
+    } catch (error) {
+      err = handleErrors(error);
+    }
+  } else {
+    try {
+      transformedConfig.active = false;
+      const response = await channelMap[lmsType].post(transformedConfig);
+      onSubmit(formFields);
+    } catch (error) {
+      err = handleErrors(error);
+    }
+  }
+  if (err) {
+    errHandler(err);
+  }
+  return !err;
+}
+
+export function checkForDuplicateNames(existingConfigNames: string[], existingData: any): FormFieldValidation {
+  const configNames: string[] = existingConfigNames?.filter((name) => name !== existingData.displayName);
+  const displayNameValidation: FormFieldValidation = {
+    formFieldId: 'displayName',
+    validator: () => {
+      return configNames?.includes('displayName')
+        ? INVALID_NAME
+        : false;
+    },
+  }
+  return displayNameValidation;
```

Comment:
Think we can simplify this logic
```suggestion
  return {
    formFieldId: 'displayName',
    validator: () => {
      return configNames?.includes(existingData.displayName)
        ? INVALID_NAME
        : false;
    },
  }
  return displayNameValidation;
```